### PR TITLE
BL-5536: Added failed notification metrics, changed logging handler, sends failed notification emails, reduced notifications processed to a 24hr period per run, updated unit & integration tests

### PR DESCRIPTION
**Pull request by Jerotire**

### CODE REVIEW

File: src/integration-test/java/uk/gov/dvsa/motr/test/integration/bouncing/email/handler/MotrBouncingEmailCleanerHandlerTest.java [hunk 1]
```diff
@@ -1,152 +0,0 @@
-package uk.gov.dvsa.motr.test.integration.bouncing.email.handler;
```
Comment (Jerotire) [inline src/integration-test/java/uk/gov/dvsa/motr/test/integration/bouncing/email/handler/MotrBouncingEmailCleanerHandlerTest.java]: Removed this integration test as the notify sendEmail process is asynchronous and causes this integration test to fail every day. Valid options are to mock Notify response (within the integration test) or poll Notify during the integration test.

**Comment by Jerotire** [comment]
Removed the integration test (https://github.com/dvsa/motr-bouncing-email-cleaner/pull/3/files#diff-a5d305d800040f9f347f8e8976463235) as the notify sendEmail process is asynchronous and causes this integration test to fail every day. Valid options are to mock Notify response (within the integration test) or poll Notify during the integration test.

File: src/main/java/uk/gov/dvsa/motr/service/EmailMessageStatusService.java [hunk 2]
```diff
@@ -20,13 +24,29 @@ public EmailMessageStatusService(NotificationClient notificationClient) {
...
+    public List<String> getEmailAddressesAssociatedWithNotifications(String status, DateTime notificationDateFilter)
+            throws NotificationClientException {
+
+        return this.getEmailsFromNotifications(this.getNotifications(status), notificationDateFilter);
+    }
+
+    /**
+     * @return a list of email addresses from notifications
+     */
+    private List<String> getEmailsFromNotifications(List<Notification> notifications, DateTime notificationDateFilter) {
```
Comment (aaronlong) [inline src/main/java/uk/gov/dvsa/motr/service/EmailMessageStatusService.java]: This private method only has one use.  We could probably just put the code in this method into getEmailAddressesAssociatedWithNotifications directly :)

File: src/main/java/uk/gov/dvsa/motr/service/NotifyService.java [hunk 3]
```diff
@@ -24,7 +25,19 @@ public SendEmailResponse sendEmail(String emailAddress, String registrationNumbe
...
+
+    public SendEmailResponse sendStatusEmail(String emailAddress, StatusReport statusReport) throws NotificationClientException {
+
+        Map<String, String> statistics = new HashMap<>();
+        statistics.put("temporary_failures", Integer.toString(statusReport.getTemporaryFailedNotifications()));
+        statistics.put("technical_failures", Integer.toString(statusReport.getTechnicalFailedNotifications()));
+        statistics.put("permanent_failures", Integer.toString(statusReport.getPermanentlyFailedNotifications()));
+        statistics.put("date", statusReport.getDate().toLocalDate().toString());
+
+        return notificationClient.sendEmail("9d900a9f-9faa-462c-bf05-63f007539474", emailAddress, statistics, "");
```
Comment (aaronlong) [inline src/main/java/uk/gov/dvsa/motr/service/NotifyService.java]: Needs to be moved into an environment variable

File: src/main/java/uk/gov/dvsa/motr/service/NotifyService.java [hunk 4]
```diff
@@ -24,7 +25,19 @@ public SendEmailResponse sendEmail(String emailAddress, String registrationNumbe
         personalisation.put("vehicle_details", registrationNumber);
```
Comment (aaronlong) [inline src/main/java/uk/gov/dvsa/motr/service/NotifyService.java]: This class needs some unit tests

File: src/main/java/uk/gov/dvsa/motr/service/SendStatusReportService.java [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import uk.gov.dvsa.motr.eventlog.EventLogger;
+import uk.gov.dvsa.motr.report.StatusReport;
+import uk.gov.service.notify.NotificationClientException;
+
+import java.util.List;
+
+public class SendStatusReportService {
```
Comment (aaronlong) [inline src/main/java/uk/gov/dvsa/motr/service/SendStatusReportService.java]: Needs unit tests

**Pull request merged**